addons/stock/static/src/stock_forecasted/stock_forecasted.js

```diff
@@ -35,22 +37,25 @@ export class StockForecasted extends Component {
             this.reloadReport();
         }
         this.warehouses = useState([]);
+        this.variants = useState([]);
 
         onWillStart(this._getReportValues);
     }
 
     async _getReportValues() {
         await this._getResModel();
-        const isTemplate = !this.resModel || this.resModel === 'product.template';
+        const isTemplate = (!this.resModel || this.resModel === 'product.template' ||
+            (this.context.active_model === 'product.template' && !this.context.variant_id));
         this.reportModelName = `stock.forecasted_product_${isTemplate ? "template" : "product"}`;
-        this.warehouses.splice(0, this.warehouses.length);
-        this.warehouses.push(...await this.orm.searchRead('stock.warehouse', [],['id', 'name', 'code']));
-        if (!this.context.warehouse_id) {
-            this.updateWarehouse(this.warehouses[0].id);
+        await this._loadWarehouses();
+
+        if (isTemplate || this.context.active_model === 'product.template') {
+            await this._loadVariants();
         }
+        const variant_id = this.context?.variant_id;
```

Comment:
```suggestion
        const variant_id = this.context.variant_id;
```
You need to define this at the start of this function, as it's also called for in the isTemplate checking.